src/zope/browserresource/tests/test_directives.py

```diff
@@ -16,49 +16,57 @@
 
 import os
 import unittest
-try:
-    from cStringIO import StringIO
-except ImportError:
-    from io import StringIO
+from io import StringIO
+
 from zope import component
-from zope.interface import Interface, implements, directlyProvides, providedBy
+from zope.interface import Interface, implementer
 
 import zope.security.management
 from zope.configuration.xmlconfig import xmlconfig, XMLConfig
 from zope.configuration.exceptions import ConfigurationError
 from zope.publisher.browser import TestRequest
-from zope.publisher.interfaces import IDefaultViewName
-from zope.publisher.interfaces.browser import IBrowserPublisher
+
+
 from zope.publisher.interfaces.browser import IBrowserRequest
-from zope.publisher.interfaces.browser import IBrowserSkinType, IDefaultSkin
-from zope.security.proxy import removeSecurityProxy, ProxyFactory
-from zope.security.permission import Permission
-from zope.security.interfaces import IPermission
+
+from zope.security.proxy import ProxyFactory
+
+
```

Comment:
Hmm...I guess that when I clean up unused imports, I mostly just tend to delete the lines in place to try to make for a cleaner diff. If I need to move an import from "below" upwards, it tends to go at the end, unless there's already a clear grouping for its package, again in the effort to make the diff cleaner. Only if it starts to look real bad (subjective, I know) do I stop to organize the whole thing.